addons/pos_sale/static/src/app/order_management_screen/sale_order_management_screen/sale_order_management_screen.js

```diff
@@ -135,32 +131,31 @@ export class SaleOrderManagementScreen extends Component {
         } catch {
             // FIXME Universal catch seems ill advised
         }
-        const order_partner = this.pos.db.get_partner_by_id(sale_order.partner_id[0]);
+        const order_partner = this.pos.models["res.partner"].get(sale_order.partner_id);
```

Comment:
this is a little confusing.
1. In the original code, `partner_id` is accessed from the `sale_order` in two ways: i) `.partner_id`; ii) `.partner_id[0]`. So there's probably a problem, unless `sale_order` becomes a different object in the method.
2. In your change, you kept the first type of accessing it `.partner_id`. Can you verify if this is what we really load from the server? I'm asking because I see from the diff in `@pos_sale/sale_order.py` that you only put `load=False` in the `read` of `sale_line`.

Reply:
Oh, I understand now. `data.read | data.searchRead` is using `load=False` by default. There is a small problem to this. It's inconsistent with the server.

When we call read or search_read from the server to prepare the data for loading, we should not forget to put `load=False` to be consistent with the UI.

Let's not do this, let's just stick with the default `load` and just explicitly put `load=False` for the moment. We'll make it implicit once we come up with a better idea.